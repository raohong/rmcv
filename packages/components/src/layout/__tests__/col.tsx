import { render, screen } from '@testing-library/react';
import { Col } from '..';

it('render correctly', () => {
  const tree = render(<Col />);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with span', () => {
  render(<Col data-testid='col' span={8} />);

  expect(screen.getByTestId('col')).toHaveStyleRule(
    'width',
    `${((8 / 24) * 100).toFixed(6)}%`,
  );
});

it('render with tag', () => {
  render(<Col data-testid='tag' component='p' />);

  expect(screen.getByTestId('tag').tagName).toBe('P');
});
