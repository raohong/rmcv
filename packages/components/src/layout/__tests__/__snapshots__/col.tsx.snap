// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  min-height: 1px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 100%;
}

<div
    class="RVuiCol-root emotion-0"
  />
</DocumentFragment>
`;
